react-basic-practice/src/features/SignUpForm/index.jsx

```diff
@@ -1,86 +1,77 @@
 import React, { useState } from 'react';
-import { Link } from 'react-router-dom';
+import { Link, useNavigate } from 'react-router-dom';
 import { v4 as uuidv4 } from 'uuid';
 
-import REGEXP from '../../constants/regexp';
-
 import Button from '../../components/common/Button';
 import Input from '../../components/common/Input';
 import Logo from '../../components/common/Logo';
 import FormGroup from '../../components/FormGroup';
 import './index.css';
 
+import { createUser, getAllUsers } from '../../services/Users';
+import Validate from '../../helpers/validate';
+import { useLoading } from '../../contexts/loading';
+
 const initialErrorMsgs = {
-  inputEmail: '',
-  inputUserName: '',
-  inputPassword: '',
+  email: '',
+  userName: '',
+  password: '',
+  form: '',
 };
 
 const initialInput = {
-  inputEmail: '',
-  inputUserName: '',
-  inputPassword: '',
+  email: '',
+  userName: '',
+  password: '',
 };
 
-
 const SignUpForm = () => {
-  const [isSignUpLoading, setIsSignUpLoading] = useState(false);
   const [errorMessage, setErrorMessage] = useState(initialErrorMsgs);
   const [inputValue, setInputValue] = useState(initialInput);
+  const navigate = useNavigate();
+  const { loading, show, hide } = useLoading();
 
   const handleInputValue = (value) => {
     setInputValue({ ...inputValue, ...value });
   };
 
-  const validation = () => {
-    for (const key in inputValue) {
-      if (inputValue[key] === '') {
-        setErrorMessage((preMsg) => ({
-          ...preMsg,
-          [key]: 'This input is required.',
-        }));
-      } else {
-        setErrorMessage((preMsg) => ({
-          ...preMsg,
-          [key]: '',
-        }));
-      }
-    }
-  };
-
-  const validateInputs = () => {
-    if (inputValue.inputPassword !== '') {
-      REGEXP.REGEXP_PASSWORD.test(inputValue.inputPassword)
-        ? setErrorMessage((preMsg) => ({ ...preMsg, inputPassword: '' }))
-        : setErrorMessage((preMsg) => ({ ...preMsg, inputPassword: 'Invalid password' }));
-    }
-
-    if (inputValue.inputEmail !== '') {
-      REGEXP.REGEXP_MAIL.test(inputValue.inputEmail)
-        ? setErrorMessage((preMsg) => ({ ...preMsg, inputEmail: '' }))
-        : setErrorMessage((preMsg) => ({ ...preMsg, inputEmail: 'Invalid email' }));
-    }
-
-    if (inputValue.inputUserName !== '') {
-      REGEXP.REGEXP_USER_NAME.test(inputValue.inputUserName)
-        ? setErrorMessage((preMsg) => ({ ...preMsg, inputUserName: '' }))
-        : setErrorMessage((preMsg) => ({ ...preMsg, inputUserName: 'Invalid user name' }));
-    }
-  };
-
-  const handleSignUp = (e) => {
+  const handleSignUp = async (e) => {
     try {
       e.preventDefault();
 
-      setIsSignUpLoading(true);
-
-      validation();
-      validateInputs();
-
-      setIsSignUpLoading(false);
+      show();
```

Comment:
`show()` you show what?

Reply:
this function to set the loading state to true

Reply:
If another developer reads your code, will she know the meaning of this function?